mqtt/mqtt-broker/tests/integration.rs

```diff
@@ -0,0 +1,87 @@
+///! This module contains tests that verify non-MQTT related functionality
+///! of the broker (like config, storage, cleanup, etc...).
+///!
+///! For tests related to MQTT protocol please use `compliance.rs`.
+use std::{any::Any, convert::Infallible};
+
+use mqtt3::{proto::ClientId, ConnectionError, Event};
+use mqtt_broker::{
+    auth::{Activity, Authorization, Authorizer},
+    BrokerBuilder, Message, SystemEvent,
+};
+use mqtt_broker_tests_util::{
+    client::TestClientBuilder,
+    server::{start_server, DummyAuthenticator},
+};
+
+/// Validates the case when authorization rules change leads to some
+/// existing sessions (connected or offline) being unauthorized and dropped.
+#[tokio::test]
+async fn drop_sessions_on_reauthorize() {
+    let broker = BrokerBuilder::default()
+        .with_authorizer(BooleanAuthorizer(true))
+        .build();
+
+    let broker_handle = broker.handle();
+    let mut server_handle = start_server(broker, DummyAuthenticator::with_id("device_1"));
+
+    // connect a client with persistent session.
+    let mut client = TestClientBuilder::new(server_handle.address())
+        .with_client_id(ClientId::IdWithExistingSession("device_1".into()))
+        .build();
+
+    // connect another client with persistent session that is always authorized.
+    let root = TestClientBuilder::new(server_handle.address())
```

Comment:
Can you name this root_client or something similar so we know it's also a client?

Reply:
Sure, I'll do that in the next pr for session cleanup.